src/sources/envoy_als/grpc.rs

```diff
@@ -33,68 +37,99 @@ impl AccessLogService for Service {
             tokio::select! {
                 _ = &mut shutdown => break,
                 stream_msg = in_stream.message() => {
-                    match stream_msg {
-                        Ok(omsg) => {
-                            match omsg {
-                                Some(msg) => {
-                                    let mut events = vec![];
+                    match self.handle_message(stream_msg, &mut stream_identifier).await {
+                        Ok(_) => {},
+                        Err(err) => {
+                            match err {
+                                StreamError::StreamClosed => break,
+                                StreamError::Grpc(status) => {
+                                    return Err(status);
+                                }
```

Comment:
What we'll actually want to do here is emit internal `Error` events for these conditions.

This is a checklist item in https://github.com/vectordotdev/vector/blob/master/docs/CONTRIBUTING.md#new-sources-sinks-and-transforms

basically in the `internal_events` directory, you'll want to create a file `envoy_als.rs` and add two new internal error events, something like `EnvoyAccessLogStreamClosed` and `EnvoyAccessLogGrpcError`. 

`internal_events/amqp.rs` has some good examples of Error internal events. (note you don't need to add modules for source/sink because there is only a source here.

Reply:
Since the stream closed one is part of normal operation, that would only leave the gRPC error. Since there's a common `GrpcError` that already exists, I've opted to just use that and not make a new internal events file for now. wdyt?

Reply:
Yes, I'm good with using the existing `GrpcError` and since the other condition is not an Error, this comment can be dismissed.